### PR TITLE
Moved camera related functionality from Frustum to Camera class

### PR DESCRIPTION
**Pull request by mvaligursky**
This PR:
- reverts this one: https://github.com/playcanvas/engine/pull/4735
- implements this suggestion: https://github.com/playcanvas/engine/pull/4735#issuecomment-1275256918

### CODE REVIEW

File: src/scene/camera.js [hunk 1]
```diff
@@ -532,6 +532,56 @@ class Camera {
...
+     * Returns an array of corners of the frustum of the camera in the local coordinate system of the camera.
+     *
+     * @param {*} near - near distance for the frustum points , defaults to the near clip distance of the camera.
+     * @param {*} far - far distance for the frustum points, defaults to the far clip distance of the camera.
+     * @returns {Array<Vec3>} - An array of corners, using a global storage space.
+     */
+    getFrustumCorners(near, far) {
+
+        near = near || this._nearClip;
+        far = far || this._farClip;
```
Comment (willeastcott) [inline src/scene/camera.js:546]: Maybe use default parameters here?

File: src/scene/camera.js [hunk 2]
```diff
@@ -532,6 +532,56 @@ class Camera {
         // ortho
         return math.clamp(sphere.radius / this._orthoHeight, 0, 1);
     }
+
+    /**
+     * Returns an array of corners of the frustum of the camera in the local coordinate system of the camera.
+     *
+     * @param {*} near - near distance for the frustum points , defaults to the near clip distance of the camera.
+     * @param {*} far - far distance for the frustum points, defaults to the far clip distance of the camera.
+     * @returns {Array<Vec3>} - An array of corners, using a global storage space.
```
Comment (willeastcott) [inline src/scene/camera.js:541]:
```suggestion
     * @param {number} [near] - Near distance for the frustum points. Defaults to the near clip distance of the camera.
     * @param {number} [far] - Far distance for the frustum points. Defaults to the far clip distance of the camera.
     * @returns {Vec3[]} - An array of corners, using a global storage space.
```

File: src/scene/camera.js [hunk 3]
```diff
@@ -532,6 +532,56 @@ class Camera {
         // ortho
         return math.clamp(sphere.radius / this._orthoHeight, 0, 1);
     }
+
+    /**
+     * Returns an array of corners of the frustum of the camera in the local coordinate system of the camera.
+     *
+     * @param {number} [near] - Near distance for the frustum points. Defaults to the near clip distance of the camera.
+     * @param {number} [far] - Far distance for the frustum points. Defaults to the far clip distance of the camera.
+     * @returns {Vec3[]} - An array of corners, using a global storage space.
```
Comment (willeastcott) [inline src/scene/camera.js:541]: Should this have `@ignore`?

File: src/scene/camera.js (hunk 1 above)
Comment (mvaligursky) [reply src/scene/camera.js:546]: yeah I always feel strange using default parameters that use `this.` - I assume it works? I'd need to test. I will in ES5 and it transpiles to what we have anyways.

Comment (mvaligursky) [reply src/scene/camera.js]: ok done, works

**Pull request merged**